src/transport/types.ts

```diff
@@ -0,0 +1,73 @@
+import events from 'events'
+import Multiaddr from 'multiaddr'
+import Connection from '../connection/connection'
+import { Sink } from '../stream-muxer/types'
+
+export type DialOptions = {
+  signal?: AbortSignal
+}
+
+/**
+ * A libp2p transport is understood as something that offers a dial and listen interface to establish connections.
+ */
+export interface Transport <DialOptions extends { signal?: AbortSignal }> {
+  new (upgrader: Upgrader, ...others: any): Transport<DialOptions>; // eslint-disable-line
```

Comment:
`...others: any` here also seems like bad idea, it should probably be either typed or made generic.